test/models/GoodsRecordsPaginationSpec.scala

```diff
@@ -0,0 +1,942 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models
+import controllers.routes
+import org.scalatest.freespec.AnyFreeSpec
+import org.scalatest.matchers.must.Matchers
+import org.scalatest.{OptionValues, TryValues}
+import uk.gov.hmrc.govukfrontend.views.Aliases.Pagination
+import uk.gov.hmrc.govukfrontend.views.viewmodels.pagination.{PaginationItem, PaginationLink}
+
+class GoodsRecordsPaginationSpec extends AnyFreeSpec with Matchers with TryValues with OptionValues {
+  private val pageSize = 10
+
+  ".getPagination" - {
+
+    "when currentPage 0" in {
+      val currentPage = 0
+      val totalPages  = 4
+      val pagination  = Pagination(
```

Comment:
These tests might all be simpler/clearer if we name it 
val expectedPagination

and then the result.blah mustEqual paginatioin.blah
could just be result mustEqual expectedPagination

Reply:
done